packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2230,7 +2230,9 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
             child: DraggableScrollableActuator(
               child: StatefulBuilder(
                 key: _currentBottomSheetKey,
-                builder: (BuildContext context, StateSetter setState) => widget.bottomSheet!,
+                builder: (BuildContext context, StateSetter setState) {
```

Comment:
Just curious, if the bottom sheet has been abruptly removed and is null at this point, should we also do anything to make the bottom sheet state closed? Should _currentBottomSheet be null? Or should the animation controller value be set to closed? 

I am not sure what the expected behavior is based on no customer test, so I am fine with landing this as is and touching base with the customer to see if this resolves their issues before making more changes. WDYT?

Reply:
Ok I think you're right about that.  I've gone back to the previous solution.

The GlobalKey error is reproducible even on master right now because of the exit animation.  But we should probably wait for feedback before we make further changes as you say.

<details>

<summary>Code to repro GlobalKey error</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return const MyHomePage(title: 'Flutter Demo Home Page');
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  final GlobalKey bottomSheetKey = GlobalKey();

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    final Widget bottomSheet = Container(
      key: bottomSheetKey,
      width: double.infinity,
      height: 100,
      color: Colors.blue,
      child: const Text('BottomSheet'),
    );

    return MaterialApp(
      theme: _counter % 2 == 0 ? ThemeData.dark() : ThemeData.light(),
      home: Scaffold(
        appBar: AppBar(
          title: Text(widget.title),
        ),
        bottomSheet: _counter % 2 == 0 ? bottomSheet : null,
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              const Text(
                'You have pushed the button this many times:',
              ),
              Text(
                '$_counter',
                style: Theme.of(context).textTheme.headlineMedium,
              ),
              if (_counter % 2 != 0) 
                bottomSheet,
            ],
          ),
        ),
        floatingActionButton: FloatingActionButton(
          onPressed: _incrementCounter,
          tooltip: 'Increment',
          child: const Icon(Icons.add),
        ),
      ),
    );
  }
}
```

</details>